### PR TITLE
Nicer error message when passing an empty configuration file

### PR DESCRIPTION
**Pull request by Kimbohlovette**
This PR checks if `cfg.Validate()` error is `errMissingReceivers` error then returns a nicely formated error.

### CODE REVIEW

File: otelcol/collector.go [hunk 1]
```diff
@@ -179,6 +180,9 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	}
 
 	if err = cfg.Validate(); err != nil {
+		if errors.Is(err, errMissingReceivers) {
+			return fmt.Errorf("no receiver configuration specified in collector configuration file")
+		}
```
Comment (TylerHelmuth) [inline otelcol/collector.go:185]: The `errMissingReceivers` is ok.  What we want to change it detecting, during `Validate` that the configuration is completely empty, not just that it is missing receiver configuration.

Comment (Kimbohlovette) [reply otelcol/collector.go:185]: Alright. Let me check inside the cfg.Validate() and  make the changes in there

File: otelcol/collector.go [hunk 2]
```diff
@@ -179,6 +180,9 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	}
 
 	if err = cfg.Validate(); err != nil {
+		if errors.Is(err, errInvalidOrMissingConfigFile) {
+			return errInvalidOrMissingConfigFile
+		}
```
Comment (TylerHelmuth) [inline otelcol/collector.go:185]:
```suggestion
```

We can return the error directly.

File: otelcol/config.go [hunk 3]
```diff
@@ -6,14 +6,16 @@ package otelcol // import "go.opentelemetry.io/collector/otelcol"
...
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/service"
 )
 
 var (
-	errMissingExporters = errors.New("no exporter configuration specified in config")
-	errMissingReceivers = errors.New("no receiver configuration specified in config")
+	errMissingExporters           = errors.New("no exporter configuration specified in config")
+	errMissingReceivers           = errors.New("no receiver configuration specified in config")
+	errInvalidOrMissingConfigFile = errors.New("missing or invalid configuration file")
```
Comment (TylerHelmuth) [inline otelcol/config.go:18]:
```suggestion
	errEmptyOrMissingConfigFile = errors.New("missing or empty configuration file")
```

File: otelcol/config.go [hunk 4]
```diff
@@ -44,6 +46,11 @@ type Config struct {
 func (cfg *Config) Validate() error {
 	// Currently, there is no default receiver enabled.
 	// The configuration must specify at least one receiver to be valid.
+
+	if cfg == nil || reflect.ValueOf(cfg).Len() < 1 {
+		return errInvalidOrMissingConfigFile
```
Comment (TylerHelmuth) [inline otelcol/config.go:51]:
```suggestion
		return errEmptyOrMissingConfigFile
```

File: otelcol/config.go [hunk 5]
```diff
@@ -44,6 +46,11 @@ type Config struct {
 func (cfg *Config) Validate() error {
 	// Currently, there is no default receiver enabled.
 	// The configuration must specify at least one receiver to be valid.
+
+	if cfg == nil || reflect.ValueOf(cfg).Len() < 1 {
```
Comment (TylerHelmuth) [inline otelcol/config.go:50]: Lets move this above the comments - those should be associate with the existing receiver check.

Comment (TylerHelmuth) [inline otelcol/config.go:50]: `cfg` is not a Array, Chan, Map, Slice, String, or pointer to Array, so this call will panic.  I believe we'll need to check the len of each of `cfg`'s sections and if they are all 0 then return this error.

File: otelcol/collector.go (hunk 2 above)
Comment (Kimbohlovette) [reply otelcol/collector.go:185]: Okay

File: otelcol/config.go (hunk 5 above)
Comment (Kimbohlovette) [reply otelcol/config.go:50]: Alright. I'm applying the changes right away

Comment (Kimbohlovette) [reply otelcol/config.go:50]: Alright no problem then. The reason why I did not want to check the length of individual properties of the cfg is that that config may grow. That's another property may be added to cfg in the future

**Comment by Kimbohlovette** [comment]
Hello @TylerHelmuth I've made the required changes. Please can you look at it?

**Review by TylerHelmuth — COMMENTED** [review]
@Kimbohlovette looks good, but I prefer the word `empty` over `invalid`.  There are lots of reasons a config could be invalid, but this check is specifically catching empty configs.

**Comment by Kimbohlovette** [comment]
> @Kimbohlovette looks good, but I prefer the word `empty` over `invalid`. There are lots of reasons a config could be invalid, but this check is specifically catching empty configs.

Okay I'm changing it to invalid

**Comment by Kimbohlovette** [comment]
Hello @TylerHelmuth I'm made that single change and also updated the branch from main

**Comment by Kimbohlovette** [comment]
@TylerHelmuth @jpkrohling. I have some tests failing but  tests ran successfully locally. Please you throw some context as to why these tests may be failing?

**Review by TylerHelmuth — APPROVED** [review]

**Comment by Kimbohlovette** [comment]
@dmitryax  please could you review my PR?

**Review by dmitryax — APPROVED** [review]

**Pull request merged**